src/Microsoft.Azure.WebJobs.Host/Triggers/TriggerData.cs

```diff
@@ -3,25 +3,26 @@
 
 using System;
 using System.Collections.Generic;
+using System.Threading.Tasks;
 using Microsoft.Azure.WebJobs.Host.Bindings;
 
 namespace Microsoft.Azure.WebJobs.Host.Triggers
 {
     /// <summary>
     /// Represents data returned after a trigger parameter value is bound.
     /// </summary>
-    public class TriggerData : ITriggerData
+    public class TriggerData : ITriggerData, IBindingDataFactory
```

Comment:
Because the binding extension APIs are based on ITriggerData and not this concrete class, I needed a pattern that worked even if an extension returned its own ITriggerData implementation. Hence the new interface.